File: app/src/main/java/com/silas/meditacao/interfaces/Extractable.java
package com.silas.meditacao.interfaces;

import com.silas.meditacao.models.Meditacao;

import java.util.ArrayList;
import java.util.Calendar;

/**
 * Created by silas on 03/05/16.
 */
public interface Extractable {
    //    encapsula o algoritmo de extração que pode variar conforme a fonte
    ArrayList<Meditacao> extraiMeditacao(Calendar day, int type);

    //    verifica se esta extraindo conteudo atual
    boolean conteudoEstaAtualizado(Calendar dia);

}

File: app/src/main/java/com/silas/meditacao/interfaces/Extractable.kt
package com.silas.meditacao.interfaces

import com.silas.meditacao.models.Meditacao


interface Extractable {
    //    encapsula o algoritmo de extração que pode variar conforme a fonte
    fun extractDevotional(): List<Meditacao>

    //    verifica se esta extraindo conteudo atual
//    fun contentIsUpdated(): Boolean

}

